File: app/src/main/java/com/iutcalendar/notification/Notif.java
package com.iutcalendar.notification;

import android.app.NotificationManager;
import android.app.PendingIntent;
import android.content.Context;
import androidx.annotation.DrawableRes;
import androidx.core.app.NotificationCompat;
import com.iutcalendar.data.DataGlobal;

public class Notif extends NotificationCompat.Builder {
    private final String chanelId;
    private final NotificationManager mNotificationManager;

    public Notif(Context context, String chanelId, String title, String msg, @DrawableRes int icon, PendingIntent pendingItent) {
        super(context, chanelId);
        this.chanelId = chanelId;
        setSmallIcon(icon);
        setContentTitle(title);
        setContentText(msg);
        setAutoCancel(true);
        setContentIntent(pendingItent);

        mNotificationManager = (NotificationManager) context.getSystemService(Context.NOTIFICATION_SERVICE);
    }

    public static void cancelAlarmNotif(Context context) {
        NotificationManager mNotificationManager = (NotificationManager) context.getSystemService(Context.NOTIFICATION_SERVICE);
        mNotificationManager.cancel(0);
    }

    public void show() {
        if (chanelId.equals(NotificationChannels.ALARM_NOTIFICATION_ID)) {
            mNotificationManager.notify(0, build());
        } else {
            mNotificationManager.notify((int) System.currentTimeMillis(), build());
        }

    }
}

File: app/src/main/java/com/iutcalendar/notification/Notif.kt
package com.iutcalendar.notification

import android.app.NotificationManager
import android.app.PendingIntent
import android.content.Context
import androidx.annotation.DrawableRes
import androidx.core.app.NotificationCompat

class Notif(context: Context, private val chanelId: String, title: String?, msg: String?, @DrawableRes icon: Int, pendingIntent: PendingIntent?) :
    NotificationCompat.Builder(context, chanelId) {
    private val mNotificationManager: NotificationManager


    init {
        setSmallIcon(icon)
        setContentTitle(title)
        setContentText(msg)
        setAutoCancel(true)
        setContentIntent(pendingIntent)
        mNotificationManager = context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
    }

    fun show() {
        mNotificationManager.notify(
            if (chanelId == NotificationChannels.ALARM_NOTIFICATION_ID) {
                0
            } else {
                System.currentTimeMillis().toInt()
            }, build()
        )
    }

    companion object {
        fun cancelAlarmNotif(context: Context) {
            (context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager).cancel(0)
        }
    }
}